src/App.tsx

```diff
@@ -1,25 +1,40 @@
 import React from 'react';
 import './App.css';
 import { getNumbers } from './utils';
+import { Pagination } from './components/Pagination';
 
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
 const items = getNumbers(1, 42).map(n => `Item ${n}`);
 
 export const App: React.FC = () => {
+  const [perPage, setPerPage] = React.useState<number>(5);
+  const [currentPage, setCurrentPage] = React.useState<number>(1);
+  const itemsListSize = items.length;
+
+  function onPerPageChange(e: React.ChangeEvent<HTMLSelectElement>): void {
+    setPerPage(+e.target.value);
+    setCurrentPage(1);
+  }
```

Comment:
It's good practice to handle state updates that are based on the previous state using a functional update. This ensures that the state update is always accurate even if there are multiple updates in quick succession.

Reply:
not valid for case